### ISSUE TITLE
PSA PAKE: Check the signature of get_password_bytes/get_user/get_peer

### ISSUE THREAD

**Issue by ronald-cron-arm** [Mbed-TLS/mbedtls#7274]
### Suggested enhancement
Check the signature of get_password_bytes/get_user/get_peer getter functions of the driver interface.

Currently we have one getter function to get the length of the password/user id/peer id and one getter function to copy the password/user id/peer id into a buffer. As the password/user id/peer id are not limited in length with this interface it is difficult for a driver to avoid a memory allocation to get the password/user id/peer id from the getter functions.

Thus why not rather having only one getter function with the following signature (user case):
 psa_status_t psa_crypto_driver_pake_get_user(
    const psa_crypto_driver_pake_inputs_t *inputs,
    const uint8_t **user_id, size_t *user_id_lentgh);
where we would just return the address of the buffer where the core stored the user identifier. 

### Justification
Mbed TLS needs this because we wants to make drivers as simple as possible.

**Comment by gilles-peskine-arm** [comment]
> I'm not sure if we want to share memory from core with the driver.

Indeed, we don't. After a driver entry point returns, a pointer to memory provided by the core is no longer guaranteed to be valid. It could be a pointer to data on the stack of a core function which then returns, or a pointer to a heap block in a managed language which gets moved by garbage collection.

This is why the driver interface only passes buffers to driver entry points, not pointers. It's up to the driver to copy the data if needed. In a few cases (IIRC only for key derivation and PAKE), the drivers can call callbacks into the core, and those callbacks can output pointers which are valid until the driver entry point returns. This is tricky to get right (the driver may not store the pointer anywhere, otherwise there is likely to be a dangling pointer at least on some platforms), so it should only be done sparingly, if it's the only way to avoid duplicating potentially large amounts of memory.

**Comment by ronald-cron-arm** [comment]
Closing as the proposal was not conform to the PSA memory model.
